swift/Picture.swift

```diff
@@ -0,0 +1,139 @@
+import CoreGraphics
+
+import thorvg
+
+/// A Swift wrapper for ThorVG's Picture, facilitating picture manipulation.
+class Picture {
+    /// Supported MIME types for loading picture data.
+    enum MimeType: String {
+        case svg = "svg"
+        case lottie = "lottie"
+        case jpg = "jpg"
+        case jpeg = "jpeg"
+        case svgxml = "svg+xml"
+        case png = "png"
+    }
+
+    /// Pointer to the underlying ThorVG picture object.
+    let pointer: OpaquePointer
+
+    /// Initializes a new Picture instance with an empty ThorVG picture object.
+    convenience init() {
+        let pointer: OpaquePointer = tvg_picture_new()
+        self.init(pointer: pointer)
+    }
+
+    /// Initializes a Picture instance from an existing ThorVG picture pointer.
+    init(pointer: OpaquePointer) {
+        self.pointer = pointer
+    }
+
+    /// Loads a picture from a given file path.
+    func load(fromPath path: String) throws {
+        guard tvg_picture_load(pointer, path) == TVG_RESULT_SUCCESS else {
+            throw ThorVGError.failedToLoadFromPath
+        }
+    }
+
+    /// Loads a picture from a given data string.
+    ///
+    /// Use the `mimeType` to indicate the data format for correct parsing.
+    func load(fromString string: String, mimeType: MimeType) throws {
+        guard let cString = string.cString(using: .utf8),
+              tvg_picture_load_data(pointer, cString, UInt32(cString.count), mimeType.rawValue, false) == TVG_RESULT_SUCCESS
+        else {
+            throw ThorVGError.failedToLoadFromDataString
+        }
+    }
+
+    /// Resizes the picture content to the given size.
+    func resize(_ size: CGSize) {
+        tvg_picture_set_size(pointer, Float(size.width), Float(size.height))
+    }
+
+    /// Retrieves the size of the picture.
+    private func getSize() -> CGSize {
+        var width: Float = 0
+        var height: Float = 0
+        tvg_picture_get_size(pointer, &width, &height)
+        return CGSize(width: Double(width), height: Double(height))
+    }
+
+    /// Applies a transformation matrix to the picture, with an optional anchor point for rotation and scaling.
+    ///
+    /// Note: the anchorPoint defaults to (0.5, 0.5), which is the center of the picture.
+    func apply(transform: CGAffineTransform, anchorPoint: CGPoint = CGPoint(x: 0.5, y: 0.5)) {
+        let size = getSize()
+
+        let pivotPoint = CGPoint(
+            x: size.width * anchorPoint.x,
+            y: size.height * anchorPoint.y
+        )
+
+        var transform = getTransform()
+        transform = transform
+            .translatedBy(x: pivotPoint.x, y: pivotPoint.y)
+            .concatenating(transform)
+            .translatedBy(x: -pivotPoint.x, y: -pivotPoint.y)
```

Comment:
This transform computation is wrong. Notice that in line 76 you're using the transform defined in line 73, not the one passed as parameter. In addition to that, I'm pretty sure you are applying the transforms in the wrong order. You should apply the existing transform, and then translate the coordinate system to the pivot point. Your code is doing the opposite. You won't notice the problem when the existing transform is the identity, but it will break for any other case. 

This is how I like to think about transforms, maybe it's helpful. Let's say you have a point and you want to apply three transformations to it (in order) defined by three matrices. Mathematically speaking, this is what you want:
```
transformedPoint = point * matrix1 * matrix2 * matrix3
```

Matrix multiplication is associative, but it is not distributive, meaning that `a * b * c == a * (b * c)` but `a * b != b * a` unless `a` or `b` is the identity matrix. So, we can rewrite the computation above as:

```
matrix = matrix1 * matrix2 * matrix3
transformedPoint = point * matrix
```

In `CoreGraphics` world, you achieve the matrix multiplication behaviour (`a * b`) by using `concatenating`. The other set of functions (`rotate(by:)`, `scaleBy`, `translateBy`) are used to get the multiplication the other way around (`b * a`). 

So, if the transformations you want to apply have a natural order, using the `concatenating` function might be clearer:

```swift
let transform = getTransform()
     // change the coordinate system to apply transforms around pivotPoint.
    .concatenating(CGAffineTransform(translationX: -pivotPoint.x, y: -pivotPoint.y))
     // apply the transform you want.
    .concatenating(transform)
    // change the coordinate system back to where it was
    .concatenating(CGAffineTransform(translationX: pivotPoint.x, y: pivotPoint.y))
``` 

You could also write the code using the `transformBy` functions, but then you would need to write the comments in the reverse order, which might not be that straightforward to understand. However, mixing the two types of functions, `concatenating` and `transformBy`, makes it harder to understand because you can't have a clear picture of the direction in which transformations are applied.

Reply:
@rlaguilar ah goodness. Great catch. Silly mistake on not using the `transform` variable!

And thank you so much for sharing how you understand `CGAffineTransform`. Helps me understand the matrix multiplication that is actually happening so well. I've updated the code to follow the `concatenating` paradigm as that seems to follow a nicer logical order.